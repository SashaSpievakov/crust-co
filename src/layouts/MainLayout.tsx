import { Outlet, useLocation } from 'react-router-dom';
import { useEffect, useRef } from 'react';

import { Wrapper, Main, Container } from '../styles/Base.styled';
import Footer from '../components/Footer/Footer';
import Header from '../components/Header/Header';

const MainLayout = () => {
  const scrollRef = useRef<null | HTMLDivElement>(null);
  const isMounted = useRef<boolean>(false);
  const location = useLocation();

  useEffect(() => {
    if (isMounted.current) {
      scrollRef.current?.scrollIntoView();
    }

    isMounted.current = true;
  }, [location.pathname]);

  return (
    <Wrapper ref={scrollRef}>
      <Header />
      <Main>
        <Container>
          <Outlet />
        </Container>
      </Main>
      <Footer />
    </Wrapper>
  );
};
export default MainLayout;
