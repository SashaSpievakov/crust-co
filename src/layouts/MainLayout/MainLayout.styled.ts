import styled from 'styled-components';

const Main = styled.main`
  padding: 40px 0;
  border-top: 1px solid ${({ theme }) => theme.colors.gray};
  border-bottom: 1px solid ${({ theme }) => theme.colors.gray};

  @media screen and (max-width: 550px) {
    padding: 30px 0;
  }
`;

export { Main };
