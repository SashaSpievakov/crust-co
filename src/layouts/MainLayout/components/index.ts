export { Footer } from './Footer';
export { Header } from './Header';
export { LayoutErrorBoundary } from './LayoutErrorBoundary';
