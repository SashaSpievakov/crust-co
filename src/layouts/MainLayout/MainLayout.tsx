import { FC } from 'react';
import { Outlet } from 'react-router-dom';

import { LayoutErrorBoundary } from '../LayoutErrorBoundary';
import { Footer } from './Footer';
import { Header } from './Header';
import { Main } from './MainLayout.styled';

export const MainLayout: FC = () => {
  return (
    <>
      <Header />
      <Main>
        <LayoutErrorBoundary>
          <Outlet />
        </LayoutErrorBoundary>
      </Main>
      <Footer />
    </>
  );
};
