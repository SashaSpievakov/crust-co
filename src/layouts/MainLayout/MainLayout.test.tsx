import '@testing-library/jest-dom';

import { rendererWithProviders } from '../../tests/helpers';
import { MainLayout } from './MainLayout';

describe('MainLayout tests', () => {
  test('should render the MainLayout component', () => {
    const snapshot = rendererWithProviders(<MainLayout />);
    expect(snapshot).toMatchSnapshot();
  });
});
