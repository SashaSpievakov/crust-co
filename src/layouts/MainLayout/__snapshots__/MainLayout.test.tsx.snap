// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout tests should render the MainLayout component 1`] = `
Object {
  "0": <header
    className="sc-bYMpWt iIjUZO"
  >
    <a
      className="sc-gYbzsP kWOdjw"
      href="/"
      ishome="home"
      onClick={[Function]}
      tabIndex={-1}
    >
      <section
        className="sc-cCjUiG gDAuXv"
      >
        <img
          alt="Pizza logo"
          height="54.5"
          src="main-logo.svg"
          width="38"
        />
        <div>
          <h1
            className="sc-cjibBx cEpyPE"
          >
            Crust & Co.
          </h1>
          <p
            className="sc-jcMfQk cLlNuA"
          >
            the most delicious pizzas in town
          </p>
        </div>
      </section>
    </a>
    <section
      className="sc-kMjNwy gbQbEk"
    >
      <svg
        className="sc-ilhmMj dNxlmF"
        data-testid="themeIcon"
        data-theme="light"
        fill="currentColor"
        height="1em"
        mode=""
        onClick={[Function]}
        onKeyDown={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        tabIndex={0}
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"
        />
      </svg>
      <a
        className="sc-bcXHqe sc-jIRcFI glVyWT hvOfFb"
        data-testid="cartLink"
        href="/cart"
        onClick={[Function]}
      >
        <span>
          0
           $
        </span>
        <div
          className="sc-hhOBVt bWENhq"
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .49.598l-1 5a.5.5 0 0 1-.465.401l-9.397.472L4.415 11H13a.5.5 0 0 1 0 1H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM3.102 4l.84 4.479 9.144-.459L13.89 4H3.102zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 1 0 2 1 1 0 0 1 0-2zm7 0a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
          />
        </svg>
        <span>
          0
        </span>
      </a>
    </section>
  </header>,
  "1": <main
    className="sc-iTFTee fOPEND"
  />,
  "2": <footer
    className="sc-lllmON gXqVeD"
  >
    <section
      className="sc-ezOQGI jrCbMM"
    >
      <p>
        Â© 2024 Crust & Co.
      </p>
      <p>
        All rights reserved.
      </p>
      <p>
        Developer - Sasha Spievakov.
      </p>
    </section>
    <section
      className="sc-iJnaPW dwsAAI"
    >
      <a
        className="sc-gikAfH dDTaNN"
        href="https://github.com/SashaSpievakov"
      >
        GitHub
      </a>
      <a
        className="sc-gikAfH dDTaNN"
        href="https://www.google.com/maps/place/687+Yonge+St,+Toronto,+ON+M4Y+2B2/@43.6689719,-79.3870519,18z/data=!3m1!4b1!4m6!3m5!1s0x882b34ae07a20ea5:0xcb9f91518efcc2c1!8m2!3d43.6689704!4d-79.3860646!16s%2Fg%2F11nnkxs6c3"
        target="_blank"
      >
        687 Yonge St
      </a>
      <a
        className="sc-gikAfH dDTaNN"
        href="mailto:spievakov@gamil.com"
      >
        spievakov@gmail.com
      </a>
    </section>
  </footer>,
}
`;
